### DISCUSSION TITLE
LSM303dlhc sensor example not compiling for nRF52840

### DISCUSSION THREAD

**Discussion by CodingGhost** [zephyrproject-rtos/zephyr#51895] [Q&A] [answered]
the LSM303dlhc example does not compile for the nRF52840 target. tried on 2.7.1 and latest main, on 2.7.1 I get
devicetree.h:305:40: error: 'DT_N_INST_0_st_lsm303dlhc_magn_P_label' undeclared (first use in this function)
and in main I get:
util.h:78:55: error: size of unnamed array is negative

**To Reproduce**
follow this guide and use this command:  west build -b nrf52840dk_nrf52840 samples/sensor/lsm303dlhc
https://docs.zephyrproject.org/latest/samples/sensor/lsm303dlhc/README.html

**Impact**
showstopper currently, as Im unable to implement the sensor.

**Environment (please complete the following information):**
 - OS: Fedora 15
 - Toolchain west

**Comment by henrikbrixandersen** [comment, marked as answer]
Well, the `nrf52840dk_nrf52840` board does not have an onboard LSM303dlhc sensor. If you added one you need to make a devicetree overlay describing how this sensor is connected to the board.

**Comment by CodingGhost** [comment]
oh. that makes sense. I will try that. I guess I can take the devicetree of the stm board as example?

**Comment by henrikbrixandersen** [comment]
> I guess I can take the devicetree of the stm board as example?

Sure.
